### PR TITLE
assert weights being 1-d tensor in bincount

### LINKED ISSUE #85777: torch.bincount crash with segmentation fault
### 🐛 Describe the bug

`torch.bincount` crash with segmentation fault when `weights` has 0 in shape.

```
import torch
torch.bincount(minlength=10, weights=torch.ones([1,0,1]), input=torch.tensor([10]))
```

output:
```
Segmentation fault (core dumped)
```

### Versions

Collecting environment information...
PyTorch version: 1.12.1+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.10

Python version: 3.7.6 (default, Jan  8 2020, 19:59:22)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-4.15.0-176-generic-x86_64-with-debian-buster-sid
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.12.1+cpu
[pip3] torchaudio==0.12.1+cpu
[pip3] torchvision==0.13.1+cpu
[conda] blas                      1.0                         mkl  
[conda] mkl                       2020.2                      256  
[conda] mkl-service               2.3.0            py37he8ac12f_0  
[conda] mkl_fft                   1.3.0            py37h54f3939_0  
[conda] mkl_random                1.1.1            py37h0573a6f_0  
[conda] numpy                     1.19.2           py37h54aff64_0  
[conda] numpy-base                1.19.2           py37hfa32c7d_0  
[conda] torch                     1.12.1+cpu               pypi_0    pypi
[conda] torchaudio                0.12.1+cpu               pypi_0    pypi
[conda] torchvision               0.13.1+cpu               pypi_0    pypi

### PR DESCRIPTION
**Pull request by yuguo68**
Summary:
as title,
fix https://github.com/pytorch/pytorch/issues/85777

Test Plan: unittest added

Differential Revision: D39913476

### CODE REVIEW

File: test/test_reductions.py [hunk 1]
```diff
@@ -1129,6 +1129,10 @@ def test_bincount(self, device):
             torch.bincount(torch.tensor([1, 3], device=device),
                            torch.tensor([.2, .2], device=device),
                            minlength=-1)
+        # n-d weights, with n > 1 throws
```
Comment (mruberry) [inline test/test_reductions.py:1132]: We should probably port these assertions to ErrorInputs on the bincount OpInfo, but this is fine for now

**Review by mruberry — APPROVED** [review]
Nice error check improvement; this should be mergeable once the tests pass

**Review by ngimel — APPROVED** [review]

**Comment by yuguo68** [comment]
test failure on `TestCommonCUDA.test_python_ref_executor__refs_fill_executor_aten_cuda_complex128` is not related.

**Pull request closed without merging**